mathesar_ui/src/components/sheet/SheetClipboardHandler.ts

```diff
@@ -107,42 +111,63 @@ export class SheetClipboardHandler<
     return this.deps.selection.getSelectedUniqueRowsId(cells);
   }
 
-  private getCopyContent(): string {
+  private getCopyContent(): { structured: string; tsv: string } {
     const cells = get(this.deps.selection.selectedCells);
     const indexedRecords = new Map(
       this.deps.getRows().map((r) => [r.rowIndex, r.record]),
     );
-    const processedColumns = this.deps.getColumnsMap();
+    const columns = this.deps.getColumnsMap();
     const recordSummaries = this.deps.getRecordSummaries();
-    const plainTextMatrix: string[][] = [];
+
+    const tsvRows: string[][] = [];
+    const structuredRows: StructuredCell[][] = [];
     for (const rowId of this.getRowIds(cells)) {
-      const plainTextRow: string[] = [];
+      const tsvRow: string[] = [];
+      const structuredRow: StructuredCell[] = [];
       for (const columnId of this.getColumnIds(cells)) {
-        if (isCellSelected(cells, { rowIndex: rowId }, { id: columnId })) {
-          const cellText = getCellText(
-            indexedRecords,
-            processedColumns,
-            rowId,
-            columnId,
-            'raw',
-            recordSummaries,
-          );
-          plainTextRow.push(cellText);
+        const column = columns.get(columnId);
+        if (!isCellSelected(cells, { rowIndex: rowId }, { id: columnId })) {
+          // Ignore cells that are not selected.
+          continue;
         }
+        if (!column) {
+          // Ignore cells with no associated column. This should never happen.
+          continue;
+        }
+        const rawCellValue = getRawCellValue(indexedRecords, rowId, columnId);
+        const formattedCellValue = getFormattedCellValue(
+          rawCellValue,
+          columns,
+          columnId,
+          recordSummaries,
+        );
+        const type = column.abstractType.identifier;
+        structuredRow.push({
+          type,
+          raw: rawCellValue,
+          formatted: formattedCellValue,
+        });
+        tsvRow.push(formattedCellValue);
       }
```

Comment:
A minor nitpick.

For cases like these, where we don't have break out of the loop, I'd prefer iterating with `forEach` and `return` statements instead of a for loop and `continue`.

I don't have a major concern towards removing the `no-continue` rule, but I'm not a fan of `continue` either.

No action item needed. I only wanted to share this.